src/main/scala/nl.knaw.dans.easy.validatebag/validation/package.scala

```diff
@@ -112,13 +117,14 @@ package object validation extends DebugEnhancedLogging {
 
   def getProfileVersion(bag: BagDir): Try[ProfileVersion] = Try {
     if ((bag / "bag-info.txt").exists) {
-      val b = bagReader.read(bag.path)
-
-      Option(b.getMetadata.get("BagIt-Profile-Version")).map(_.asScala.toList).map {
-        case (v :: _) => Try { v.toInt }.filter(profileVersionDois.keys.toSet.contains).getOrElse(0)
-        case _ => 0
-      }.getOrElse(0)
-    }
-    else 0
+      Try { bagReader.read(bag.path) } // Reading the bag will fail if bag-info.txt is malformed
+        .map { b =>
+          Option(b.getMetadata.get("BagIt-Profile-Version")).map(_.asScala.toList).map {
+            case v :: _ => Try { v.toInt }.filter(profileVersionDois.keys.toSet.contains).getOrElse(ProfileVersion0.versionNumber)
```

Comment:
You're using `ProfileVersion0.versionNumber` 5 times here. **Challenge:** can you reduce this number?